### PR TITLE
feature: create l10n-extract script

### PR DESCRIPTION
**Pull request by beyackle**
## Description

While formatMessage is a good way to extract strings from the JavaScript parts of Composer, we did not have a way to extract strings from the schema files. This provides a new script to extract just the relevant parts of a schema and produce something we can send off to be translated without a translator needing to see the rest of the schema. The intent is for these translated files to get merged in as plugins.

The l10n-extract script will take a single file (in JSON format, just like our schemas are) and produce a stripped-down version of it consisting only of the fields "title", "subtitle", "label", and "description" and any object containing another object with at least one of those fields. The new file will be placed in the same place, with ".en-US.json" appended to its name.

## Task Item

refs #3809 

## Screenshots

image

### CODE REVIEW

**Comment by a-b-r-o-w-n** [comment]
Also, I wonder if this PR should include the server support for loading the locale uischemas.

**Comment by beyackle** [comment]
The schema file contains text like the sidebar's various titles and descriptions, so those fields will need to be localized. Loading a localized schema from the server is something I'd like to handle in a future PR; this one is just providing a convenient way to extract all the fields we care about.
image

File: Composer/packages/server/schemas/locales/en-US/sdk.uischema [hunk 1]
```diff
@@ -0,0 +1,326 @@
+{
```
Comment (a-b-r-o-w-n) [inline Composer/packages/server/schemas/locales/en-US/sdk.uischema:1]: The naming convention for these will just be `sdk.en-us.uischema`

**Comment by beyackle** [comment]
Superseded by #3913.

**Pull request closed without merging**